### PR TITLE
feat(drain): add disable-ha arg to core-agent

### PR DESCRIPTION
**Pull request by Abhinandan-Purkait**
- Add the HA_ENABLED env to core-agent with {{ .Values.agents.ha.enabled }}

### CODE REVIEW

**Review by avishnu — APPROVED** [review]
lgtm

**Comment by avishnu** [comment]
Wondering if it will be a good idea to pass this flag to csi-node as well? Behavior could be, if csi-node detects that ANA is not supported, it should CLBO unless HA_ENABLED is set to false.
@tiagolobocastro @Abhinandan-Purkait ?

**Comment by Abhinandan-Purkait** [comment]
> Wondering if it will be a good idea to pass this flag to csi-node as well? Behavior could be, if csi-node detects that ANA is not supported, it should CLBO unless HA_ENABLED is set to false. @tiagolobocastro @Abhinandan-Purkait ?

Yeah, we can do that.
Only downside maybe is we will be restricting normal operations which can still happen without HA.

**Review by niladrih — APPROVED** [review]

**Comment by tiagolobocastro** [comment]
> Wondering if it will be a good idea to pass this flag to csi-node as well? Behavior could be, if csi-node detects that ANA is not supported, it should CLBO unless HA_ENABLED is set to false. @tiagolobocastro @Abhinandan-Purkait ?

If Ha is not enabled then agent-ha-node is not running anyway so it will not trigger any ha operations

**Comment by avishnu** [comment]
> > Wondering if it will be a good idea to pass this flag to csi-node as well? Behavior could be, if csi-node detects that ANA is not supported, it should CLBO unless HA_ENABLED is set to false. @tiagolobocastro @Abhinandan-Purkait ?
> 
> If Ha is not enabled then agent-ha-node is not running anyway so it will not trigger any ha operations

I'm suggesting for csi-node, not agent-ha. Kind of fail-fast approach, user will have to then set ha_enabled = false to proceed with normal operations.

File: chart/templates/mayastor/agents/core/agent-core-deployment.yaml [hunk 1]
```diff
@@ -63,7 +63,7 @@ spec:
             - name: RUST_LOG
               value: {{ .Values.agents.core.logLevel }}
             - name: HA_ENABLED
-              value: {{ .Values.agents.ha.enabled }}
+              value: "{{ .Values.agents.ha.enabled }}"
```
Comment (niladrih) [inline chart/templates/mayastor/agents/core/agent-core-deployment.yaml:66]: Maybe this?
```suggestion
              value: {{ .Values.agents.ha.enabled | quote }}
```

**Review by niladrih — COMMENTED** [review]
Left a minor comment

**Review by niladrih — APPROVED** [review]

**Review by niladrih — COMMENTED** [review]
Left one comment

File: chart/templates/mayastor/agents/core/agent-core-deployment.yaml [hunk 2]
```diff
@@ -57,6 +57,7 @@ spec:
             - "--volume-commitment={{ .Values.agents.core.capacity.thin.volumeCommitment }}"{{ if .Values.agents.core.partialRebuildWaitPeriod }}
             - "--faulted-child-wait-period={{ .Values.agents.core.partialRebuildWaitPeriod }}"{{ end }}{{ if .Values.eventing.enabled }}
             - "--events-url=nats://{{ .Release.Name }}-nats:4222"{{ end }}
+            - "--ha-enabled={{ .Values.agents.ha.enabled }}"
```
[the remark this answers is not in the document]
Comment (Abhinandan-Purkait) [reply chart/templates/mayastor/agents/core/agent-core-deployment.yaml:60]: Yeah, if we go with `--disable-ha` we can just pass it. I will make that change.

[the remark this answers is not in the document]
Comment (niladrih) [reply chart/templates/mayastor/agents/core/agent-core-deployment.yaml:60]: Yeah. Additionally, `--flag=true` is not a thing with clap, I think. Those only work with int/string/path, etc. Bools can only be set based on if the flag exists or not.

[the remark this answers is not in the document]
Comment (Abhinandan-Purkait) [reply chart/templates/mayastor/agents/core/agent-core-deployment.yaml]: Done.

**Review by niladrih — APPROVED** [review]

**Review by tiagolobocastro — APPROVED** [review]

**Comment by Abhinandan-Purkait** [comment]
bors merge

**Pull request merged**